File: rec-core/src/main/java/net/kimleo/rec/repository/selector/expr/SelectorTokenType.java
package net.kimleo.rec.repository.selector.expr;

public enum SelectorTokenType {
    ID("ID"),
    DOT("."),

    LEFT_SQUARE("]"),
    RIGHT_SQUARE("]"),
    COMMA(","),

    AS("as");

    String repr;

    SelectorTokenType(String repr) {
        this.repr = repr;
    }
}

File: src/main/kotlin/net/kimleo/rec/repository/selector/expr/SelectorTokenType.kt
package net.kimleo.rec.repository.selector.expr

enum class SelectorTokenType(val repr: String) {
    ID("ID"),
    DOT("."),
    LEFT_SQUARE("["),
    RIGHT_SQUARE("]"),
    PIPE("|"),
    EQUAL("="),
    COMMA(","),
    IS("is"),
    MATCHES("matches"),
}